CI: Test_MergeRoutePolicies/both_set_standard_and_large_communities_with_differing_local_preference is flaky

## CI failure

```
--- FAIL: Test_MergeRoutePolicies (0.00s)
--- FAIL: Test_MergeRoutePolicies/both_set_standard_and_large_communities_with_differing_local_preference (0.00s)

    policies_test.go:486: 
        	Error Trace:	/host/pkg/bgpv1/manager/reconcilerv2/policies_test.go:486
        	Error:      	Not equal: 
        	            	expected: &types.RoutePolicy{Name:"policy", Type:0, Statements:[]*types.RoutePolicyStatement{(*types.RoutePolicyStatement)(0xc000a70360), (*types.RoutePolicyStatement)(0xc000a703f0)}}
        	            	actual  : &types.RoutePolicy{Name:"policy", Type:0, Statements:[]*types.RoutePolicyStatement{(*types.RoutePolicyStatement)(0xc000a70900), (*types.RoutePolicyStatement)(0xc000a70870)}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -4,2 +4,24 @@
        	            	  Statements: ([]*types.RoutePolicyStatement) (len=2) {
        	            	+  (*types.RoutePolicyStatement)({
        	            	+   Conditions: (types.RoutePolicyConditions) {
        	            	+    MatchNeighbors: ([]string) (len=1) {
        	            	+     (string) (len=7) "fd00::3"
        	            	+    },
        	            	+    MatchPrefixes: ([]*types.RoutePolicyPrefixMatch) <nil>,
        	            	+    MatchFamilies: ([]types.Family) (len=1) {
        	            	+     (types.Family) {
        	            	+      Afi: (types.Afi) 2,
        	            	+      Safi: (types.Safi) 0
        	            	+     }
        	            	+    }
        	            	+   },
        	            	+   Actions: (types.RoutePolicyActions) {
        	            	+    RouteAction: (types.RoutePolicyAction) 1,
        	            	+    AddCommunities: ([]string) <nil>,
        	            	+    AddLargeCommunities: ([]string) (len=1) {
        	            	+     (string) (len=14) "3333:3333:3333"
        	            	+    },
        	            	+    SetLocalPreference: (*int64)(100)
        	            	+   }
        	            	+  }),
        	            	   (*types.RoutePolicyStatement)({
        	            	@@ -33,24 +55,2 @@
        	            	    }
        	            	-  }),
        	            	-  (*types.RoutePolicyStatement)({
        	            	-   Conditions: (types.RoutePolicyConditions) {
        	            	-    MatchNeighbors: ([]string) (len=1) {
        	            	-     (string) (len=7) "fd00::3"
        	            	-    },
        	            	-    MatchPrefixes: ([]*types.RoutePolicyPrefixMatch) <nil>,
        	            	-    MatchFamilies: ([]types.Family) (len=1) {
        	            	-     (types.Family) {
        	            	-      Afi: (types.Afi) 2,
        	            	-      Safi: (types.Safi) 0
        	            	-     }
        	            	-    }
        	            	-   },
        	            	-   Actions: (types.RoutePolicyActions) {
        	            	-    RouteAction: (types.RoutePolicyAction) 1,
        	            	-    AddCommunities: ([]string) <nil>,
        	            	-    AddLargeCommunities: ([]string) (len=1) {
        	            	-     (string) (len=14) "3333:3333:3333"
        	            	-    },
        	            	-    SetLocalPreference: (*int64)(100)
        	            	-   }
        	            	   })
        	Test:       	Test_MergeRoutePolicies/both_set_standard_and_large_communities_with_differing_local_preference
```

Recent example: https://github.com/cilium/cilium/actions/runs/12652309900/job/35255153384

Looks like the flaky test was introduced in https://github.com/cilium/cilium/pull/36414 /cc @dswaffordcw

--- a/pkg/bgpv1/manager/reconcilerv2/policies_test.go
+++ b/pkg/bgpv1/manager/reconcilerv2/policies_test.go
@@ -4,6 +4,7 @@
 package reconcilerv2
 
 import (
+	"slices"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -483,8 +484,25 @@ func Test_MergeRoutePolicies(t *testing.T) {
 				req.NoError(err)
 			}
 
+			if tt.expected != nil {
+				req.NotNil(result)
+				SortRouteStatementsByName(result.Statements)
+				SortRouteStatementsByName(tt.expected.Statements)
+			}
+
 			req.Equal(tt.expected, result)
 		})
 	}
+}
 
+func SortRouteStatementsByName(statements []*types.RoutePolicyStatement) {
+	slices.SortFunc(statements, func(i, j *types.RoutePolicyStatement) int {
+		if i.Conditions.String() < j.Conditions.String() {
+			return -1
+		} else if i.Conditions.String() > j.Conditions.String() {
+			return 1
+		} else {
+			return 0
+		}
+	})
 }

